Installation of espflash fails with error

Error message:
```
error[E0658]: exclusive range pattern syntax is experimental
  --> /home/georgik/.cargo/registry/src/github.com-1ecc6299db9ec823/espflash-1.1.0/src/chip/esp32/esp32c3.rs:85:53
   |
85 |             (ImageFormatId::DirectBoot, None | Some(3..)) => {
```

Reason: Version of Rust 1.54. Required version 1.55.

--- a/Install-RustToolchain.ps1
+++ b/Install-RustToolchain.ps1
@@ -16,13 +16,14 @@ $ErrorActionPreference = "Stop"
 $ProgressPreference = 'SilentlyContinue'
 $ExportContent = ""
 #Set-PSDebug -Trace 1
+$RustcMinimalMinorVersion="55"
 
 "Processing configuration:"
 "-InstalltationMode    = ${InstallationMode}"
 "-ToolchainVersion     = ${ToolchainVersion}"
 "-ToolchainDestination = ${ToolchainDestination}"
 
-function InstallRustup() {
+function InstallRust() {
     Invoke-WebRequest https://win.rustup.rs/x86_64 -OutFile rustup-init.exe
     ./rustup-init.exe --default-toolchain stable -y
     $env:PATH+=";$env:USERPROFILE\.cargo\bin"
@@ -34,11 +35,17 @@ function InstallRustFmt() {
 }
 
 if (-Not (Get-Command rustup -ErrorAction SilentlyContinue)) {
-    InstallRustup
+    InstallRust
 }
 
 if ((rustup show | Select-String -Pattern stable).Length -eq 0) {
-    rustup toolchain install stable
+    InstallRust
+}
+
+if ((rustc --version).Substring(8,2) -lt $RustcMinimalMinorVersion) {
+    "rustc version is too low, requires 1.${RustcMinimalMinorVersion}"
+    "calling rustup"
+    InstallRust
 }
 
 # It seems there is a dependency on nightly for some reason only on Windows

--- a/install-rust-toolchain.sh
+++ b/install-rust-toolchain.sh
@@ -7,6 +7,7 @@ if [ -z "${RUSTUP_HOME}" ]; then
 fi
 TOOLCHAIN_DESTINATION_DIR="${RUSTUP_HOME}/toolchains/esp"
 
+RUSTC_MINIMAL_MINOR_VERSION="55"
 INSTALLATION_MODE="install" # reinstall, uninstall
 CLEAR_DOWNLOAD_CACHE="NO"
 EXTRA_CRATES="cargo-pio espflash ldproxy"
@@ -89,6 +90,15 @@ set -e
 
 # Check required tooling - rustc, rustfmt
 command -v rustup || install_rust
+
+# Check minimal rustc version
+RUSTC_MINOR_VERSION=`rustc --version | sed -e 's/^rustc 1\.\([^.]*\).*/\1/'`
+if [ "${RUSTC_MINOR_VERSION}" -lt "${RUSTC_MINIMAL_MINOR_VERSION}" ]; then
+    echo "rustc version is too low, requires 1.${RUSTC_MINIMAL_MINOR_VERSION}"
+    echo "calling rustup"
+    install_rust
+fi
+
 command -v cargo || source_cargo
 rustup toolchain list | grep stable || install_rust_toolchain stable
 rustup toolchain list | grep nightly || install_rust_toolchain nightly

